### DISCUSSION TITLE
Is there a way to create/inject custom email templates?

### DISCUSSION THREAD

**Discussion by nabbl** [unchainedshop/unchained#335] [Q&A] [answered]
Hi,

I can see that there are some mail templates in `/platform/templates/` however I can see no way to customize them without changing the unchained platform package as well?

Any hint on how to do that properly?

greetings
Daniel

**Comment by schmidsi** [comment, marked as answer]
Hi, sorry for the late reply. Yes you can configure the E-Mail templates like this:

https://github.com/unchainedshop/thisisnotacommercial.com/blob/master/unchained/boot.js#L5

https://github.com/unchainedshop/thisisnotacommercial.com/blob/master/unchained/templates/index.js

This is using an old engine, but as far as I know, the mailing did not change much.

**Comment by schmidsi** [comment]
Better docs hopefully coming soon 😊

**Reply by pozylon** [reply]
@schmidsi No it's not working like this anymore, more fancy now.

@nabbl a MESSAGING template is defined like
```
const accountActionTemplate = (props) => {
  return [
    {
      type: 'EMAIL',
      input: {
        from: 'noreply@xxx.ch',
        to: 'info@xxx.ch',
        subject: `title`,
        text: "blabla",
        attachments: [{ filename: '', path: ''}],
      },
    },
  ];
};
```

Then you can wire that function like:
```
import { MessagingDirector } from 'meteor/unchained:core-messaging';
import { MessageTypes } from 'meteor/unchained:platform';

MessagingDirector.configureTemplate(
    MessageTypes.ACCOUNT_ACTION,
    accountActionTemplate,
  );
```

This is where the defaults are:
https://github.com/unchainedshop/unchained/blob/master/packages/platform/setup-templates.js
https://github.com/unchainedshop/unchained/blob/master/packages/platform/templates/account-action.js

When you overwrite a message type like account action, make sure you run configureTemplate after startPlatform on boot

**Reply by nabbl** [reply]
merci vielmals!
das hilft enorm

**Comment by Mikearaya** [comment]
@nabbl thanks for your interest. based on your request we have added the documentation on how to do this. https://docs.unchained.shop/config/email-template/

**Reply by nabbl** [reply]
thank you! really helpful
